### PR TITLE
[metallb] Add metallb module usage example

### PR DESCRIPTION
**Pull request by konstantin-axenov**
Signed-off-by: Konstantin Aksenov <konstantin.aksenov@flant.com>

## Description

- Add metallb module usage example
- Small ingress usage refactoring. Renamed general example

## Why do we need it, and what problem does it solve?

This isn't clear what does `general example` means for ingress controller. And we have nice metallb module, but no examples. It's a shame.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: metallb
type: chore
summary: Add metallb module usage example
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @konstantin-axenov. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @konstantin-axenov. I've started the workflow for label 'e2e/run/gcp'...

File: ee/modules/380-metallb/docs/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+title: "The metallb module: usage"
+---
+
+{% raw %}
+
+Metallb can be used in Static (Bare Metal) clusters when you can't order a load balancer from a cloud provider. Metallb can work in L2 or BGP modes. Below is an example of Metallb usage in L2 mode.
+We will create Ingress Controller with `LoadBalancer` inlet. And we will also expose standalone nginx using service with `LoadBalancer` type.
+First, you have to decide, which NodeGroups will be used to deploy applications that have to be exposed by LoadBalancer service.
+Ingress controllers run on frontend nodes and nginx runs on worker node in this example. They have common label `node-role/metallb=""`.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:10]:
```suggestion
Metallb can be used in Static (Bare Metal) clusters when you can't order a load balancer from a cloud provider. Metallb can work in L2 or BGP modes. Below is an example of Metallb usage in L2 mode.

We will create an Ingress Controller with `LoadBalancer` inlet. And we will also expose standalone Nginx using a Service with the `LoadBalancer` type.

First, you have to decide which NodeGroups will be used to deploy applications that have to be exposed by the LoadBalancer service.
Ingress controllers run on frontend nodes, and Nginx runs on a worker node in this example. They have common label `node-role/metallb=""`.
```

File: ee/modules/380-metallb/docs/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+spec:
+  disruptions:
+    approvalMode: Manual
+  nodeTemplate:
+    labels:
+      node-role/metallb: ""
+  nodeType: Static
+```
+
+Check that nodes have correct labels.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:43]:
```suggestion
Check that nodes have the correct labels.
```

File: ee/modules/380-metallb/docs/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+```bash
+kubectl get nodes -l node-role/metallb
+NAME              STATUS   ROLES      AGE   VERSION
+demo-frontend-0   Ready    frontend   61d   v1.21.14
+demo-frontend-1   Ready    frontend   61d   v1.21.14
+demo-worker-0     Ready    worker     61d   v1.21.14
+```
+
+Module `metallb` is disabled by default, so you have to explicitly enable it. You also have to set correct `nodeSelector` and `tolerations` for Metallb speakers.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:53]:
```suggestion
Module `metallb` is disabled by default, so you have to explicitly enable it. You also have to set the correct `nodeSelector` and `tolerations` for Metallb speakers.
```

File: ee/modules/380-metallb/docs/USAGE.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  inlet: LoadBalancer
+  nodeSelector:
+    node-role.deckhouse.io/frontend: ""
+  tolerations:
+  - effect: NoExecute
+    key: dedicated.deckhouse.io
+    value: frontend
+```
+
+Check that service with type `LoadBalancer` is created in namespace `d8-ingress-nginx`.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:91]:
```suggestion
Check that service with the type `LoadBalancer` is created in the namespace `d8-ingress-nginx`.
```

File: ee/modules/380-metallb/docs/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+```shell
+kubectl -n d8-ingress-nginx get svc main-load-balancer 
+NAME                 TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
+main-load-balancer   LoadBalancer   10.222.255.194   192.168.199.100   80:30236/TCP,443:32292/TCP   30s
+```
+
+Your ingress controller is accessible on external ip address.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:99]:
```suggestion
Your Ingress controller is accessible on an external IP address.
```

File: ee/modules/380-metallb/docs/USAGE.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+```shell
+curl -s -o /dev/null -w "%{http_code}" 192.168.199.100
+404
+```
+
+Expose our standalone nginx on `8080` port.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:106]:
```suggestion
Expose your standalone Nginx on `8080` port.
```
I wouldn't use `your`, `ours` etc., but `yours` has been used before already.

File: ee/modules/380-metallb/docs/USAGE.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+Check service.
+
+```shell
+kubectl get svc nginx
+NAME    TYPE           CLUSTER-IP     EXTERNAL-IP       PORT(S)          AGE
+nginx   LoadBalancer   10.222.9.190   192.168.199.101   8080:31689/TCP   3m11s
+```
+
+Now you can access application using curl.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE.md:121]:
```suggestion
Now you can access the application using curl.
```

File: ee/modules/380-metallb/docs/USAGE_RU.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
+---
+title: "The metallb module: пример конфигурации"
+---
+
+{% raw %}
+
+Metallb можно использовать в статических (Bare Metal) кластерах, когда вы не можете заказать load balancer у облачного провайдера. Metallb может работать в L2 или BGP режиме. Ниже представлен пример использования Metallb в L2 режиме.
+Мы создадим Ingress Controller с инлетом `LoadBalancer`. А также дадим доступ к отдельно запущенному nginx, используя сервис с типом `LoadBalancer`.
+Во-первых, необходимо определить, какие NodeGroup'ы будут использоваться для запуска приложений, к которым будет предоставлен доступ.
+В этом примере ingress контроллеры запускаются на frontend узлах, а nginx на worker узле. У всех узлов есть общий лейбл `node-role/metallb=""`.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE_RU.md:10]:
```suggestion
Metallb можно использовать в статических (Bare Metal) кластерах, когда вы не можете заказать load balancer у облачного провайдера. Metallb может работать в L2 или BGP режиме.

Ниже представлен пример использования Metallb в L2 режиме.
Мы создадим Ingress Controller с inlet `LoadBalancer`. А также дадим доступ к отдельно запущенному Nginx, используя сервис с типом `LoadBalancer`.

Во-первых, необходимо определить, какие NodeGroup'ы будут использоваться для запуска приложений, к которым будет предоставлен доступ.
В этом примере Ingress-контроллеры запускаются на frontend-узлах, а Nginx — на worker-узле. У всех узлов есть общий лейбл `node-role/metallb=""`.
```

File: ee/modules/380-metallb/docs/USAGE_RU.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+spec:
+  disruptions:
+    approvalMode: Manual
+  nodeTemplate:
+    labels:
+      node-role/metallb: ""
+  nodeType: Static
+```
+
+Проверьте, что на нодах проставлен корректный лейбл
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE_RU.md:43]:
```suggestion
Проверьте, что на узлах проставлен корректный лейбл.
```

File: ee/modules/380-metallb/docs/USAGE_RU.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+Проверьте, что сервис с типом `LoadBalancer` создан в namespace `d8-ingress-nginx`.
+
+```shell
+kubectl -n d8-ingress-nginx get svc main-load-balancer 
+NAME                 TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
+main-load-balancer   LoadBalancer   10.222.255.194   192.168.199.100   80:30236/TCP,443:32292/TCP   30s
+```
+
+Ваш ingress контроллер доступен по внешнему ip адресу.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE_RU.md:99]:
```suggestion
Ваш Ingress-контроллер доступен по внешнему IP-адресу.
```

File: ee/modules/380-metallb/docs/USAGE_RU.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+```shell
+curl -s -o /dev/null -w "%{http_code}" 192.168.199.100
+404
+```
+
+Теперь предоставим доступ к nginx на порту `8080`.
```
Comment (z9r5) [inline ee/modules/380-metallb/docs/USAGE_RU.md:106]:
```suggestion
Теперь предоставим доступ к Nginx на порту `8080`.
```

File: modules/402-ingress-nginx/docs/USAGE_RU.md [hunk 12]
```diff
@@ -96,8 +73,26 @@ spec:
...
+  inlet: HostPort
+  hostPort:
+    httpPort: 80
+    httpsPort: 443
+    behindL7Proxy: true
+```
+
 ## Пример для Bare metal (MetalLB Load Balancer)
 
+Модуль MetalLB на текущий момент доступен только в EE редакции.
```
Comment (z9r5) [inline modules/402-ingress-nginx/docs/USAGE_RU.md:94]:
```suggestion
Модуль MetalLB на текущий момент доступен только в редакции Enterprise Edition.
```

File: modules/402-ingress-nginx/docs/USAGE.md [hunk 13]
```diff
@@ -96,7 +73,25 @@ spec:
...
+  inlet: HostPort
+  hostPort:
+    httpPort: 80
+    httpsPort: 443
+    behindL7Proxy: true
+```
+
+## An example for Bare metal (MetalLB Load Balancer)
+
+MetalLB is currently available only in EE version.
```
Comment (z9r5) [inline modules/402-ingress-nginx/docs/USAGE.md:94]:
```suggestion
The `metallb` module is currently available only in the Enterprise Edition version.
```

**Review by z9r5 — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**